bosh-director/lib/bosh/director/deployment_plan/update_config.rb

```diff
@@ -95,6 +100,19 @@ def parse_watch_times(value)
       def serial?
         !!@serial
       end
+
+      private
+      def parse_arguments(value, size)
+        return value if size.nil?
+        case value
+          when /^\d+%$/
+            [((/\d+/.match(value)[0].to_i * size) / 100).round, size].min
+          when /\A[-+]?[0-9]+\z/
+            value.to_i
+          else
+            raise 'cannot be converted'
+        end
+      end
```

Comment:
This method has two possible return types, string or int, depending on the context. This seems like a big recipe for possible future confusion. We noticed lower down that there are tests that explicitly call out this strange behavior, but we're unsure why it's necessary in the first place. Can we make sure that all return values from `UpdateConfig.canaries(size)` are a single type?